### PR TITLE
Clarify some corners of the segmenter documentation

### PR DESCRIPTION
**Pull request by eggrobin**
Some of these were brought up by @robertbastian.

Related to #3261 (but does not address all questions brought up there).

### CODE REVIEW

**Review by robertbastian — COMMENTED** [review]
Great, however I cannot review this for correctness

**Review by aethanyc — APPROVED** [review]
Thanks for clarifying these documentation! Looks good to me. 

I've spotted some link format errors. Please correct them before merging.

File: experimental/segmenter/src/indices.rs [hunk 1]
```diff
@@ -2,7 +2,11 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-/// Similar to CharIndices for Latin-1 character
+/// Similar to [`core::str::CharIndices`] for Latin-1 strings, represented as [`[u8]`].
```
[the remark this answers is not in the document]
Comment (eggrobin) [reply experimental/segmenter/src/indices.rs:5]: This one is intentional, the strings are represented by `[u8]` (it is the code units that are represented by `u8`).

File: experimental/segmenter/src/indices.rs [hunk 2]
```diff
@@ -30,7 +34,12 @@ impl<'a> Iterator for Latin1Indices<'a> {
     }
 }
 
-/// Similar to CharIndices for UTF-16 character
+/// Similar to [`core::str::CharIndices`] for UTF-16 strings, represented as `[u16]`.
```
[the remark this answers is not in the document]
Comment (eggrobin) [reply experimental/segmenter/src/indices.rs:37]: This one is intentional, the strings are represented by `[u16]` (it is the code units that are represented by `u16`).

File: experimental/segmenter/src/indices.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (aethanyc) [reply experimental/segmenter/src/indices.rs:5]: Ah, you're right! Sorry for the wrong suggestion.

**Review by aethanyc — APPROVED** [review]

File: experimental/segmenter/src/indices.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (aethanyc) [reply experimental/segmenter/src/indices.rs:5]: Nit: probably should get rid of the outer bracket `[]` since `[u8]` doesn't link to a doc.

**Review by sffc — APPROVED** [review]
thank you!

**Comment by eggrobin** [comment]
(re-adding Robert for UTC+2 rubberstamping)

**Review by robertbastian — APPROVED** [review]

**Pull request merged**